front/pages/w/[wId]/data-sources/vaults/[vaultId]/categories/[category]/data_source_views/[dataSourceViewId].tsx

```diff
@@ -67,19 +68,23 @@ export const getServerSideProps = withDefaultUserAuthRequirements<
     };
   }
 
+  const vault = dataSourceView.vault;
+  const canWrite = vault.canWrite(auth);
+
   return {
     props: {
       category: context.query.category as DataSourceViewCategory,
       dataSource: dataSourceView.dataSource.toJSON(),
       dataSourceView: dataSourceView.toJSON(),
       gaTrackingId: config.getGaTrackingId(),
       isAdmin,
+      canWrite,
```

Comment:
Outside of the server code let's be a bit more explicit
```suggestion
      canWriteInVault,
```